server/entity/effect/effect.go

```diff
@@ -74,6 +89,12 @@ func (e Effect) Level() int {
 	return e.lvl
 }
 
+// WithDuration returns a new effect with the given duration.
+func (e Effect) WithDuration(d time.Duration) Effect {
```

Comment:
This method also doesn't apply for instant effects, it should be possible to do this a different way.

Reply:
To elaborate on that, say you do have an `effect.Effect` value, you could just create a new effect with the same level and type but with a different duration.